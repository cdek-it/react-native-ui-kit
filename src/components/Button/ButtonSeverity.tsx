import { memo } from 'react'

import { BaseButton } from './BaseButton'
import { useSeverityButtonStyles } from './styles'
import type { ButtonProps, ButtonSeverityProps } from './types'

/**
 * Button component
 * @param size - button size
 * @param shape - button shape
 * @param loading - button loading state
 * @param variant - button variant
 * @param disabled - button disabled state
 * @param iconOnly - button with only Icon
 * @param iconPosition - icon position
 * @param Icon - Tabler icon
 * @param label - button label
 * @param style - external style control for component
 * @param severity - severity button styling variant
 * @see BaseButton
 */
export const ButtonSeverity = memo<ButtonProps & ButtonSeverityProps>(
  ({ severity, ...props }) => {
    const buttonStyles = useSeverityButtonStyles(severity)

    return <BaseButton {...props} {...buttonStyles} />
  }
)
