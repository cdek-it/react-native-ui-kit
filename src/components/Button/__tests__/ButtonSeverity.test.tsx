import { render } from '@testing-library/react-native'

import { ButtonSeverity } from '../ButtonSeverity'
import type { ButtonSeverityProps } from '../types'

describe('ButtonSeverity component tests', () => {
  const buttonSnapshotCases = generatePropsCombinations<ButtonSeverityProps>({
    severity: ['help', 'info', 'warning', 'danger', 'success'],
  })

  test.each(buttonSnapshotCases)('severity - $severity', (props) => {
    const renderedButton = render(<ButtonSeverity {...props} label='Button' />)

    expect(renderedButton.toJSON()).toMatchSnapshot()
  })
})
