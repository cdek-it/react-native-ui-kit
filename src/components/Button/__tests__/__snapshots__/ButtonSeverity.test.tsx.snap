// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonSeverity component tests severity - danger 1`] = `
<View
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#e26e63",
      "borderColor": "rgba(255, 255, 255, 0.0001)",
      "borderRadius": 7,
      "borderWidth": 1,
      "flexDirection": "row",
      "gap": 7,
      "height": 35,
      "justifyContent": "center",
      "maxHeight": 35,
      "minHeight": 35,
      "paddingHorizontal": 14,
      "paddingVertical": 0,
    }
  }
>
  <Text
    style={
      {
        "color": "#ffffff",
        "fontSize": 14,
        "fontWeight": 700,
        "includeFontPadding": false,
        "verticalAlign": "middle",
      }
    }
    testID="Button_Text"
  >
    Button
  </Text>
</View>
`;

exports[`ButtonSeverity component tests severity - help 1`] = `
<View
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#ab77df",
      "borderColor": "rgba(255, 255, 255, 0.0001)",
      "borderRadius": 7,
      "borderWidth": 1,
      "flexDirection": "row",
      "gap": 7,
      "height": 35,
      "justifyContent": "center",
      "maxHeight": 35,
      "minHeight": 35,
      "paddingHorizontal": 14,
      "paddingVertical": 0,
    }
  }
>
  <Text
    style={
      {
        "color": "#ffffff",
        "fontSize": 14,
        "fontWeight": 700,
        "includeFontPadding": false,
        "verticalAlign": "middle",
      }
    }
    testID="Button_Text"
  >
    Button
  </Text>
</View>
`;

exports[`ButtonSeverity component tests severity - info 1`] = `
<View
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#49a2ef",
      "borderColor": "rgba(255, 255, 255, 0.0001)",
      "borderRadius": 7,
      "borderWidth": 1,
      "flexDirection": "row",
      "gap": 7,
      "height": 35,
      "justifyContent": "center",
      "maxHeight": 35,
      "minHeight": 35,
      "paddingHorizontal": 14,
      "paddingVertical": 0,
    }
  }
>
  <Text
    style={
      {
        "color": "#ffffff",
        "fontSize": 14,
        "fontWeight": 700,
        "includeFontPadding": false,
        "verticalAlign": "middle",
      }
    }
    testID="Button_Text"
  >
    Button
  </Text>
</View>
`;

exports[`ButtonSeverity component tests severity - success 1`] = `
<View
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#68c170",
      "borderColor": "rgba(255, 255, 255, 0.0001)",
      "borderRadius": 7,
      "borderWidth": 1,
      "flexDirection": "row",
      "gap": 7,
      "height": 35,
      "justifyContent": "center",
      "maxHeight": 35,
      "minHeight": 35,
      "paddingHorizontal": 14,
      "paddingVertical": 0,
    }
  }
>
  <Text
    style={
      {
        "color": "#ffffff",
        "fontSize": 14,
        "fontWeight": 700,
        "includeFontPadding": false,
        "verticalAlign": "middle",
      }
    }
    testID="Button_Text"
  >
    Button
  </Text>
</View>
`;

exports[`ButtonSeverity component tests severity - warning 1`] = `
<View
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#ffb330",
      "borderColor": "rgba(255, 255, 255, 0.0001)",
      "borderRadius": 7,
      "borderWidth": 1,
      "flexDirection": "row",
      "gap": 7,
      "height": 35,
      "justifyContent": "center",
      "maxHeight": 35,
      "minHeight": 35,
      "paddingHorizontal": 14,
      "paddingVertical": 0,
    }
  }
>
  <Text
    style={
      {
        "color": "#ffffff",
        "fontSize": 14,
        "fontWeight": 700,
        "includeFontPadding": false,
        "verticalAlign": "middle",
      }
    }
    testID="Button_Text"
  >
    Button
  </Text>
</View>
`;
